src/network/network_admin.cpp

```diff
@@ -791,6 +796,61 @@ NetworkRecvStatus ServerNetworkAdminSocketHandler::Receive_ADMIN_EXTERNAL_CHAT(P
 	return NETWORK_RECV_STATUS_OKAY;
 }
 
+/***********
+ * Authentication receive and send functions
+ ************/
+
+NetworkRecvStatus ServerNetworkAdminSocketHandler::Receive_ADMIN_JOIN(Packet *p)
+{
+	if (this->status != ADMIN_STATUS_INACTIVE) return this->SendError(NETWORK_ERROR_NOT_EXPECTED);
+
+	this->admin_name = p->Recv_string(NETWORK_CLIENT_NAME_LENGTH);
+	this->admin_version = p->Recv_string(NETWORK_REVISION_LENGTH);
+	uint16_t protocol_mask = p->Recv_uint16();
+
+	if (this->admin_name.empty() || this->admin_version.empty()) {
+		/* no name or version supplied */
+		return this->SendError(NETWORK_ERROR_ILLEGAL_PACKET);
+	}
+
+	/* Mask out any protocols that are too new for us. */
+	protocol_mask &= (1 << ADMIN_AUTH_METHOD_END) - 1;
+	if (_settings_client.network.admin_public_keys.empty()) ClrBit(protocol_mask, ADMIN_AUTH_METHOD_X25519_AUTHORIZED_KEY);
+
+	if (protocol_mask == 0) {
+		/* There is no way to authorize this admin, with the protocols the admin and we both support. */
+		return this->SendError(NETWORK_ERROR_NOT_AUTHORIZED);
```

Comment:
It wouldn't hurt to return here to the client the ciphers didn't align. That is valuable information during debugging.

Reply:
My thought was that there's no authorized method that can be used. The nasty thing is that we are reusing the "game" error codes enumeration here, and it would generally be an unused error with a translation that no-one is ever going to see in-game unless they look at the console with debug net>=1.

Sure, I can add an error, but they're not as cheap as save game versions.